addons/web_editor/static/src/js/editor/transcoder.js

```diff
@@ -229,9 +246,30 @@ function classToStyle($editable) {
         } else {
             $target.attr('style', style);
         }
+        // Apple Mail
         if (this.nodeName === 'TD' && !this.childNodes.length) {
             this.innerHTML = '&nbsp;';
         }
+
+        // Outlook
+        if (this.nodeName === 'A' && $target.hasClass('btn') && !$target.children().length) {
+            var $hack = $('<table class="o_outlook_hack"><tr><td></td></tr></table>');
+            $hack.find('td')
+                .attr('height', $target.outerHeight())
+                .css('margin', $target.css('padding'))
+                .css('border-radius', $target.css('border-radius'))
+                .css('background-color', $target.css('background-color'));
+            $target.after($hack);
+            $target.appendTo($hack.find('td'));
+            var node = $hack[0].previousSibling;
+            if (node && node.nodeType === 3 && !node.textContent.match(/\S/)) {
```

Comment:
`Node.TEXT_NODE`
(and again: why is this needed ?)

Reply:
because, the are some time a white space unbreakable and this char add a line when is a table and is invisible when it's a link